src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -424,6 +392,19 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
     return refreshedToken;
   }
 
+  protected get headers(): HttpHeaders {
+    const optionHeaders: { [key: string]: string | string[] } = this.getOption('headers') ?? {};
+    let headers = this.buildAuthHeader() || new HttpHeaders();
+
+    headers = headers.append('Content-Type', 'application/x-www-form-urlencoded');
+
+    Object.entries(optionHeaders).forEach(([key, value]) => {
+      headers = headers.append(key, value);
+    });
+
+    return headers;
+  }
```

Comment:
Could we move a common parts between oauth and pass strategies to the base class?
```
  protected get headers(): HttpHeaders {
    const optionHeaders: { [key: string]: string | string[] } = this.getOption('headers') ?? {};
    let headers = this.buildAuthHeader() || new HttpHeaders();

    headers = headers.append('Content-Type', 'application/x-www-form-urlencoded');

    Object.entries(optionHeaders).forEach(([key, value]) => {
      headers = headers.append(key, value);
    });

    return headers;
  }
```